pkg/kube/controllers/quarksstatefulset/quarksstatefulset_reconciler.go

```diff
@@ -108,28 +110,18 @@ func (r *ReconcileQuarksStatefulSet) Reconcile(request reconcile.Request) (recon
 	}
 
 	// Get the current StatefulSet.
-	currentStatefulSet, currentVersion, err := GetMaxStatefulSetVersion(ctx, r.client, qStatefulSet)
+	_, currentVersion, err := GetMaxStatefulSetVersion(ctx, r.client, qStatefulSet)
 	if err != nil {
 		return reconcile.Result{}, ctxlog.WithEvent(qStatefulSet, "StatefulSetNotFound").Error(ctx, "Could not retrieve latest StatefulSet owned by QuarksStatefulSet '", request.NamespacedName, "': ", err)
 	}
 
 	// Calculate the desired statefulSets
-	desiredStatefulSets, desiredVersion, err := r.calculateDesiredStatefulSets(qStatefulSet, currentVersion)
+	desiredStatefulSets, _, err := r.calculateDesiredStatefulSets(qStatefulSet, currentVersion)
```

Comment:
Do we still need the currentVersion here

Reply:
We "only" need them for the annotation. It is not needed, but quite nice to have because it is an easy way to see if the stateful set has been updated. I refactored it a bit to be not that confusing.